flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/enumerator/tracker/SplitTracker.java

```diff
@@ -159,12 +170,35 @@ public boolean isAssigned(String splitId) {
         return assignedSplits.contains(splitId);
     }
 
+    /**
+     * Function to get children splits available for given parent ids. This will ensure not to
+     * iterate all the values in knownSplits so saving compute
+     */
+    public List<DynamoDbStreamsShardSplit> splitsAvailableForAssignment(
```

Comment:
nit: can we name something meaningful like `getUnassignedChildSplits`

Reply:
Let's also add some unit tests for these public methods!

Reply:
will do